components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -222,6 +226,14 @@ public static void validateDomain(String domainName) throws Exception {
                 throw new TenantManagementClientException(ERROR_CODE_EMPTY_EXTENSION);
             }
         }
+        String regex = CommonUtil.getTenantDomainRegexPattern();
+        if (StringUtils.isNotBlank(regex)) {
+            if (!isFormatCorrect(regex, domainName)) {
+                throw new TenantManagementClientException(ERROR_CODE_TENANT_DOES_NOT_MATCH_REGEX_PATTERN.getCode(),
+                        String.format(ERROR_CODE_TENANT_DOES_NOT_MATCH_REGEX_PATTERN.getMessage(), domainName, regex));
+            }
+            return;
```

Comment:
Don't we need to execute the remaining lines if the TenantDomainRegexPattern defined ?

Reply:
other validations are check for `.` character and illegal characters. We can combine all of them in this regex pattern and validate at a single step